src/homework8/arraylist/ArrayList.java

```diff
@@ -0,0 +1,107 @@
+package homework8.arraylist;
+
+import java.util.Arrays;
+
+public class ArrayList<T> implements List<T> {
+    private int position = 0;
+    private T[] array;
+
+    public ArrayList() {
+        array = (T[]) new Object[10];
+    }
+
+    @Override
+    public void add(T value) {
+        if (position < array.length) {
+            array[position] = value;
+            position++;
+        } else {
+            int newArrayLength = array.length + (array.length >> 1);
+            array = Arrays.copyOf(array, newArrayLength);
+            array[position] = value;
+            position++;
+        }
+    }
+
+    @Override
+    public void add(T value, int index) {
+        try {
+            T[] tempArray = (T[]) new Object[array.length + 1];
+            System.arraycopy(array, 0, tempArray, 0, index);
+            tempArray[index] = value;
+            System.arraycopy(array, index, tempArray, index + 1, array.length - index);
+            array = tempArray;
+        } catch (IndexOutOfBoundsException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
```

Comment:
how will work this method if list will be empty

Reply:
Ok. I understood.